### ISSUE TITLE
Errror Happen when use Batch_Normal Layer? 

### ISSUE THREAD

**Issue by yuantangliang** [dmlc/mxnet#3184]
when  i use batch_norm layer in network, it fail when the first epoch end. the error message list below.How can I fixed it? 

code like below:
`            
data = mx.symbol.Convolution(data=data, kernel=self.kernel_size, num_filter=self.feature_count,
                                         name=self.name, pad=self.pad)`
` data = mx.symbol.BatchNorm(data=data,name=self.name+"batch_normal")`

report errors:
 _checkpoint to "output/data_normal_deepyannet/data_normal_deepyannet-0001.params"
[10:44:59] /home/meter/Desktop/mxnet/dmlc-core/include/dmlc/logging.h:235: [10:44:59] src/operator/./cudnn_batch_norm-inl.h:138: Check failed: (cudnnBatchNormalizationForwardInference(s->dnn_handle_, CUDNN_BATCHNORM_SPATIAL, &a, &b, io_desc_, x.dptr_, io_desc_, y.dptr_, mean_desc_, gamma.dptr_, beta.dptr_, moving_mean.dptr_, moving_inv_var.dptr_, param_.eps)) == (CUDNN_STATUS_SUCCESS) 
[10:44:59] /home/meter/Desktop/mxnet/dmlc-core/include/dmlc/logging.h:235: [10:44:59] src/engine/./threaded_engine.h:306: [10:44:59] src/operator/./cudnn_batch_norm-inl.h:138: Check failed: (cudnnBatchNormalizationForwardInference(s->dnn_handle_, CUDNN_BATCHNORM_SPATIAL, &a, &b, io_desc_, x.dptr_, io_desc_, y.dptr_, mean_desc_, gamma.dptr_, beta.dptr_, moving_mean.dptr_, moving_inv_var.dptr_, param_.eps)) == (CUDNN_STATUS_SUCCESS) 
An fatal error occurred in asynchronous engine operation. If you do not know what caused this error, you can try set environment variable MXNET_ENGINE_TYPE to NaiveEngine and run with debugger (i.e. gdb). This will force all operations to be synchronous and backtrace will give you the series of calls that lead to this error. Remember to set MXNET_ENGINE_TYPE back to empty after debugging.
terminate called after throwing an instance of 'dmlc::Error'
  what():  [10:44:59] src/engine/./threaded_engine.h:306: [10:44:59] src/operator/./cudnn_batch_norm-inl.h:138: Check failed: (cudnnBatchNormalizationForwardInference(s->dnn_handle_, CUDNN_BATCHNORM_SPATIAL, &a, &b, io_desc_, x.dptr_, io_desc_, y.dptr_, mean_desc_, gamma.dptr_, beta.dptr_, moving_mean.dptr_, moving_inv_var.dptr_, param_.eps)) == (CUDNN_STATUS_SUCCESS) 
An fatal error occurred in asynchronous engine operation. If you do not know what caused this error, you can try set environment variable MXNET_ENGINE_TYPE to NaiveEngine and run with debugger (i.e. gdb). This will force all operations to be synchronous and backtrace will give you the series of calls that lead to this error. Remember to set MXNET_ENGINE_TYPE back to empty after debugging._

**Comment by yuantangliang** [comment]
When I set batch_size below 1024,everything is OK,but when the batch_size is bigger than 1024,this error happens.

**Comment by piiswrong** [comment]
Sounds like a cudnn issue. cudnn probably doesn't support batchsize > 1024. I suggest you ask on nvidia forum
